tdct/deconvolution.py

```diff
@@ -0,0 +1,864 @@
+import sys
+import os
+
+import tifffile
+if getattr(sys, 'frozen', False):
+    # programm runs in a bundle (pyinstaller)
+    execdir = sys._MEIPASS
+else:
+    execdir = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(execdir)
+
+import numpy as np
+from PyQt5 import QtWidgets
+from . import clrmsg
+from . import TDCT_debug
+
+try:
+    import tifffile as tf
+except:
+    sys.exit("Please install tifffile, e.g.: pip install tifffile")
+
+debug = TDCT_debug.debug
+
+import scipy.signal
+import scipy.fft
+
+#Internal helper class to handle progress bar
+class _progrBarHandle():
+    def __init__(self, qtprocessbar, niter):
+        self.iter=0
+        self.niter = niter
+        self.qtbar = qtprocessbar
+
+        # if self.qtbar:
+        # 	self.qtbar.value= self.iter/self.niter*100
+        # 	QtWidgets.QApplication.processEvents()
+        self.updateUI() #Not working
+    def increment(self):
+        self.iter+=1
+        self.updateUI()
+    def updateUI(self):
+        if self.qtbar:
+            self.qtbar.setValue(self.iter/self.niter*100)
+            QtWidgets.QApplication.processEvents()
+    def setmax(self):
+        self.iter= self.niter
+        self.updateUI()
+
+# def _convertAndNormalise(d0):
+#     d1= d0.astype('float32')
+#     vmax= d0.max()
+#     vmin=d0.min()
+#     d2 = (d1-vmin)/(vmax-vmin)
+#     return d2
+
+def _convertAndNormalise(d0, doNormalise=True):
+    d1= d0.astype('float32')
+    ret=d1
+    if doNormalise:
+        vmax= d0.max()
+        vmin=d0.min()
+        d2 = (d1-vmin)/(vmax-vmin)
+        ret=d2
+    return ret
+
+def _centered(arr, newshape):
+    # Return the center newshape portion of the array.
+    newshape = np.asarray(newshape)
+    currshape = np.array(arr.shape)
+    startind = (currshape - newshape) // 2
+    endind = startind + newshape
+    myslice = [slice(startind[k], endind[k]) for k in range(len(endind))]
+    return arr[tuple(myslice)]
+
+def doRLDeconvolution(datapath , psfdatapath , niter=0, qtprocessbar=None):
+    #Internal class to handle progress bar
+
+    ''' This code is not giving correct results
+    This is probably because the usage of the psf flipped
+    In a later version doRLDeconvolution5 (based on DeconvolutionLab2 instead of using psf-flipped in the 2nd convolution,
+    a correlation is performed. '''
+
+    #Estimate progress iterations
+    nProgrIter = 2*niter + 6 #Check if ok
```

Comment:
Again capitalisation of variable. I think it's best to use the python conventions for naming but consistency is also good and sometimes you use camel case for variables (as here) and other times python convention (as in some variables below). This could be n_progr_iter for example.